### PR TITLE
randomness-simulator

### PR DESCRIPTION
**Pull request by kaisbaccour**

### CODE REVIEW

File: src/simulator.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+/// use nois::{coinflip, Side,randomness_simulator};
+/// use cosmwasm_std::testing::mock_env;
+///
+/// let env = mock_env();
+/// let result = coinflip(randomness_simulator(env).to_array().unwrap());
+/// assert_eq!(result, Side::Tails);
+///
+/// ```
+///
+pub fn randomness_simulator(env: Env) -> HexBinary {
```
Comment (webmaster128) [inline src/simulator.rs:24]: Also let's take a reference here such that the user does not need to clone env:

```suggestion
pub fn randomness_simulator(env: &Env) -> [u8; 32] {
```

File: src/simulator.rs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    let block_height = env.block.height;
+    let mut hasher = Sha256::new();
+    hasher.update((block_height).to_be_bytes());
+    let hash: [u8; 32] = hasher.finalize().into();
+    // Should we make here some safety before unsafely unwrapping? Is there really a risk here? In that case the function would return a result instead?
+    HexBinary::try_from(hash).unwrap()
```
[the remark this answers is not in the document]
Comment (kaisbaccour) [reply src/simulator.rs]: sticking  to sha after discussion on discord

File: src/simulator.rs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+/// let env = mock_env();
+/// let result = coinflip(randomness_simulator(&env));
+/// assert_eq!(result, Side::Tails);
+///
+/// ```
+///
+pub fn randomness_simulator(env: &Env) -> [u8; 32] {
+    let block_height = env.block.height;
+    let mut hasher = Sha256::new();
+    hasher.update((block_height).to_be_bytes());
```
Comment (webmaster128) [inline src/simulator.rs:27]:
```suggestion
    hasher.update(block_height.to_be_bytes());
```

**Review by webmaster128 — COMMENTED** [review]
cool stuff

**Pull request merged**